pymc3/examples/MLDA_multilevel_groundwater_flow.py

```diff
@@ -0,0 +1,413 @@
+# before running the notebook.
+# 
+# The files within `./mlda` contain the following:
+# 
+# `GwFlow.py`: This module is a FEniCS implementation of steady state groundwater flow in a confined aquifer,
+# an elliptic Partial Differential Equation (PDE). It creates a mesh on a unit square domain, given the input
+# resolution, imposes fixed head boundary conditions of 1 and 0 at left and right boundaries, respectively,
+# and no-flow boundary conditions at the top and bottom boundaries. It then solves the problem given some FEniCS [3]
+# function, representative of heterogeneous, isotropic hydraulic conductivity.
+# 
+# `RandomProcess.py`: This module generates realisations of a Gaussian Random Field with a squared exponential
+# covariance function, as described in [2], representative of the (unknown) aquifer conductivity field. It takes a n
+# x d matrix of nodal coordinates, a covariance length scale and a truncation parameter k, and computes the k
+# largest Karhunen–Loève (KL) eigenmodes of the covariance matrix. It then generates random field realisations
+# according to the mean and standard deviation of the desired log-conductivity field and a vector of random parameters.
+# 
+# `Model.py`: A module that wraps the groundwater flow solver and random process into one object. When the
+# solve-method is called, it will solve the groundwater flow problem given a vector of random field parameters. The
+# user can then extract hydraulic heads at datapoints using the get_data-method.
+# 
+# 
+# ### Dependencies
+# The code has been developed and tested with Python 3.6. You will need to have pymc3 installed and
+# also install [FEniCS](https://fenicsproject.org/) for your system.
+#
+#
+# ### References
+# [1] Dodwell, Tim & Ketelsen, Chris & Scheichl, Robert & Teckentrup, Aretha. (2019). Multilevel
+# Markov Chain Monte Carlo. SIAM Review. 61. 509-545. https://doi.org/10.1137/19M126966X
+# 
+# [2] Scarth, C., Adhikari, S., Cabral, P. H., Silva, G. H. C., & Prado, A. P. do. (2019). Random field simulation
+# over curved surfaces: Applications to computational structural mechanics. Computer Methods in Applied Mechanics and
+# Engineering, 345, 283–301. https://doi.org/10.1016/j.cma.2018.10.026
+# 
+# [3] The FEniCS Project Version 1.5 M. S. Alnaes, J. Blechta, J. Hake, A. Johansson, B. Kehlet, A. Logg,
+# C. Richardson, J. Ring, M. E. Rognes and G. N. Wells, Archive of Numerical Software, vol. 3, 2015
+# 
+
+# ### Import modules
+
+# In[1]:
+
+
+# Import groundwater flow model utils
+import sys
+sys.path.insert(1, '../../docs/source/notebooks/mlda/')
+
+
+# In[2]:
+
+
+import os
+import numpy as np
+import time
+import pymc3 as pm
+import theano.tensor as tt
+from Model import Model, model_wrapper, project_eigenpairs
+from itertools import product
+import matplotlib.pyplot as plt
+os.environ['OPENBLAS_NUM_THREADS'] = '1'  # Set environmental variable
+
+# ### Set parameters
+
+# In[ ]:
```

Comment:
Might want to remove these comments, looks like an artefact from copying from a notebook?